Packages/Blockchain/Sources/Tron/TronService.swift

```diff
@@ -312,11 +420,82 @@ extension TronService: ChainSyncable {
 
 extension TronService: ChainStakable {
     public func getValidators(apr: Double) async throws -> [DelegationValidator] {
-        return []
+        let systemUnstacking = DelegationValidator.system(
+            chain: chain,
+            name: "Unstaking"
+        )
+
+        return try await provider
+            .request(.listwitnesses)
+            .map(as: WitnessesList.self).witnesses
+            .map {
+                DelegationValidator(
+                    chain: chain,
+                    id: try addressBase58(hex: $0.address),
+                    name: .empty,
+                    isActive: $0.isJobs ?? false,
+                    commision: 0,
+                    apr: 0
+                )
+            } + [systemUnstacking]
     }
 
     public func getStakeDelegations(address: String) async throws -> [DelegationBase] {
-        fatalError()
+        async let getAccount = account(address: address)
+        async let getReward = reward(address: address)
+        async let getVlidators = getValidators(apr: 0)
+
+        let (account, reward, validators) = try await (
+            getAccount,
+            getReward,
+            getVlidators
+        )
+
+        let pendingDelegations: [DelegationBase] = (account.unfrozenV2 ?? []).compactMap { unfrozen in
+            guard let expireTime = unfrozen.unfreeze_expire_time,
+                  let amount = unfrozen.unfreeze_amount else { return nil }
+
+            let completionDate = Date(timeIntervalSince1970: TimeInterval(expireTime / 1000))
+            let balance = BigInt(amount)
+
+            return DelegationBase(
+                assetId: chain.assetId,
+                state: Date() < completionDate ? .pending : .awaitingWithdrawal,
+                balance: BigInt(balance).description,
+                shares: .empty,
+                rewards: .zero,
+                completionDate: completionDate,
+                delegationId: completionDate.description,
+                validatorId: DelegationValidator.systemId
+            )
+        }
+
+        let votes = account.votes ?? []
+        let totalVotes = votes.reduce(0) { $0 + $1.vote_count }
+
+        let delegations: [DelegationBase] = votes.compactMap { vote -> DelegationBase? in
+            guard let validator = validators.first(where: { $0.id == vote.vote_address }) else { return nil }
+
+            let rewards: String = {
+                guard totalVotes > 0 else { return String(reward) }
+                let proportion = Double(vote.vote_count) / Double(totalVotes)
+                let rewardForVote = (Double(reward) * proportion)
+                return String(format: "%.0f", rewardForVote)
```

Comment:
Can we convert to `BigInt()` and then pass to `DelegationBase()` to make sure it's always big int value